packages/jupyter-ai/jupyter_ai/extension.py

```diff
@@ -527,20 +512,57 @@ def _init_chat_handlers(self, ychat: YChat) -> Dict[str, BaseChatHandler]:
                     continue
 
             if command_name in chat_handlers:
-                self.log.error(
-                    f"Unable to register chat handler `{chat_handler.id}` because command `{command_name}` already has a handler"
+                self.log.warn(
+                    f"Overriding existing handler `{command_name}` with `{chat_handler.id}`."
                 )
-                continue
 
-            # The entry point is a class; we need to instantiate the class to send messages to it
-            chat_handlers[command_name] = chat_handler(**chat_handler_kwargs)
+            # Special handling for native `/ask`
+            if (
+                command_name == "/ask"
+                and chat_handler.__module__ == "jupyter_ai.chat_handlers.ask"
+            ):
+                try:
+                    learn_ep = next(
+                        (ep for ep in chat_handler_eps if ep.name == "learn"), None
+                    )
+                    if not learn_ep:
+                        self.log.error(
+                            "No entry point found for 'learn' handler; skipping '/ask' registration."
+                        )
+                        continue
+
+                    LearnChatHandler = learn_ep.load()
+                    learn_handler = LearnChatHandler(**chat_handler_kwargs)
+
+                    # Lazy import Retriever
+                    from jupyter_ai.chat_handlers.learn import Retriever
```

Comment:
Can this be moved to a top-level import? I don't see an obvious benefit to doing the import here.

Reply:
I don't think that this specific import statement is significant, so I think it would make more sense to keep the PR minimal and easily portable to `2.x`.

We can bring back this change if we discover that this statement does make a significant impact on the import times.